f600233d1b154d507b782f84604bcb12/update/sys_script_include_f563b62d47152110f39f7601e36d43de.xml

```diff
@@ -15,137 +15,202 @@ following locations:&#13;
         <name>HothProvisioning</name>
         <script><![CDATA[var HothProvisioning = Class.create();
 HothProvisioning.prototype = {
-  PORTFOLIO_TABLE: "x_g_dis_atat_portfolio",
-  TASK_ORDER_TABLE: "x_g_dis_atat_task_order",
-  OPERATOR_TABLE: "x_g_dis_atat_operator",
-  CLIN_TABLE: "x_g_dis_atat_clin",
-
-  portfolio: null,
-  errUtil: new ErrorHandler(),
-
-  initialize: function(portfolioRecord) {
-    this.portfolio = portfolioRecord;
-  },
-  /**
-   * Get all CLINs associated with a Task Order
-   *
-   * @param {string} taskOrderSysId
-   * @return {Object[]} an array of CLINs
-   */
-  getClins: function(taskOrderSysId) {
-    try {
-      var clinTableFields = [
-        "clin_number", "pop_start_date", "pop_end_date", "classification_level", "type", 
-      ];
+	PORTFOLIO_TABLE: "x_g_dis_atat_portfolio",
+	TASK_ORDER_TABLE: "x_g_dis_atat_task_order",
+	OPERATOR_TABLE: "x_g_dis_atat_operator",
+	CLIN_TABLE: "x_g_dis_atat_clin",
+
+	portfolio: null,
+	errUtil: new ErrorHandler(),
+
+	initialize: function(portfolioRecord) {
+		if (!portfolioRecord) {
+			throw this.errUtil.createError(
+				"Portfolio must not be null.",
+				this.errUtil.INVALID_INPUT
+			);
+		}
+		else {
+			this.portfolio = portfolioRecord;
+		}
+	},
+
+	convertClassificationToHothFormat: function(snowClassification) {
 		var classificationMap = {
 			U: "UNCLASSIFIED",
 			S: "SECRET",
 			TS: "TOP_SECRET"
 		};
+		return classificationMap[snowClassification];
+	},
 
-      return new global.GlideQuery(this.CLIN_TABLE)
-        .where("task_order", taskOrderSysId)
-        .select(clinTableFields)
-        .toArray(20)
-        .map(removeSysId)
-		.map(function(clin) {
-		  clin.classification_level = classificationMap[clin.classification_level];
-		  return clin;
-	  })
-
-    } catch (error) {
-      throw this.errUtil.createError(
-        "HothProvisioning --> getClins(): " + error,
-        this.errUtil.METHOD_ERROR
-      );
-    }
-  },
-
-  /**
+	convertClassificationToSnowFormat: function(hothClassification) {
+		var classificationMap = {
+			"UNCLASSIFIED" : "U",
+			"SECRET": "S",
+			"TOP_SECRET": "TS"
+		};
+		return classificationMap[hothClassification];
+	},	
+
```

Comment:
Wish we had the foresight to avoid this, however the new more complete solution and function names are much improved